specutils/io/default_loaders/wcs_fits.py

```diff
@@ -85,9 +90,80 @@ def wcs1d_fits_loader(file_obj, spectral_axis_unit=None, flux_unit=None,
     if not isinstance(file_obj, fits.hdu.hdulist.HDUList):
         hdulist.close()
 
+        if wcs.naxis > 4:
+            raise ValueError('FITS file input to wcs1d_fits_loader is > 4D')
+        elif wcs.naxis > 1:
+            for i in range(wcs.naxis - 1, 0, -1):
+                try:
+                    wcs = wcs.dropaxis(i)
+                except(_wcs.NonseparableSubimageCoordinateSystemError) as e:
+                    raise ValueError(f'WCS cannot be reduced to 1D: {e} {wcs}')
+
     return Spectrum1D(flux=data, wcs=wcs, meta=meta)
 
 
+@custom_writer("wcs1d-fits")
+def wcs1d_fits_writer(spectrum, file_name, update_header=False, **kwargs):
+    """
+    Write spectrum with spectral axis defined by its WCS to (primary)
+    IMAGE_HDU of a FITS file.
+
+    Parameters
+    ----------
+    spectrum: Spectrum1D
+    file_name: str
+        The path to the FITS file
+    update_header: bool
+        Update FITS header with all compatible entries in `spectrum.meta`
+    unit : str or `~astropy.units.Unit`
+        Unit for the flux (and associated uncertainty)
+    dtype : str or `~numpy.dtype`
+        Floating point type for storing flux array
+    """
+    # Create HDU list from WCS
+    try:
+        wcs = spectrum.wcs
+        hdulist = wcs.to_fits()
+        header = hdulist[0].header
+    except AttributeError as err:
+        raise ValueError(f'Only Spectrum1D objects with valid WCS can be written as wcs1d: {err}')
+
+    # Verify spectral axis constructed from WCS
+    wl = spectrum.spectral_axis
+    dwl = (wcs.all_pix2world(np.arange(len(wl)), 0) - wl.value) / wl.value
```

Comment:
`all_pix2world` seems to be a FITS-specific method -- I can easily imagine that users are going to want to write out their spectrum loaded from e.g. an ascii file (and therefore have a gwcs), so it seems this writer is only useful for `Spectrum1D` objects that are built with FITS wcs's?

Reply:
More generally it is a method of `astropy.wcs`'s `WCS`, but that is based on the FITS model and constructed from a FITS header. This is already tested with the `wcs.to_fits()` call above, but the error message could perhaps state this more specifically (I am afraid it will still potentially confuse users why their `spectrum.wcs` does not qualify).
Generally the `gwcs.WCS`, at least in the use cases here, primarily seems to function as a container for an array of `spectral_axis` values, so it cannot be directly used here. For linear, equidistant spectral axes it should be relatively straightforward to construct an `astropy.wcs.WCS` that encodes this `spectral_axis`, but I don't think this should be part of a loader - it would better fit into `utils.wcs_utils`.

Reply:
#676 brought up another issue with how and where to handle WCS properties:
One straightforward fix would not work for some of our example files which do not include a canonical wavelength unit definition at all, e.g.  `L5g_0355+11_Cruz09.fits`.

As I see the current situation, `Spectrum1D` no longer accepts `flux` or `spectral_axis` without units, and the `gwcs` also always has one, but using the `wcs.WCS` still allows the loader to sneak around this requirement.

So my proposal would be to add some additional checks for a spectral unit in this loader and assign a default unit, if none is found (which generally will be `Angstrom` for FITS spectra) – either in this PR or a follow-up. But the question remains if this verification should be left in general to the loader, or if `Spectrum1D` should not also check this if initialised with `wcs` and no `spectral_axis`.

Reply:
> As I see the current situation, Spectrum1D no longer accepts flux or spectral_axis without units, and the gwcs also always has one, but using the wcs.WCS still allows the loader to sneak around this requirement

This isn't strictly true: if a fits wcs is provided, the initializer will still attempt to create a `SpectralAxis` from the wcs information and fall back to a length unit if unsuccessful:

```python
In [13]: my_wcs = fitswcs.WCS(header={'CDELT1': 1, 'CRVAL1': 6562.8,
    ...:                                  'CTYPE1': 'WAVE', 'RESTFRQ': 1400000000, 'CRPIX1': 25},
    ...:                          naxis=1)
    ...: spec = Spectrum1D(flux=[5,6,7] * u.Jy, wcs=my_wcs)
WARNING: FITSFixedWarning: 'datfix' made the change 'Set DATE-REF to '1858-11-17' from MJD-REF'. [astropy.wcs.wcs]
WARNING:astropy:FITSFixedWarning: 'datfix' made the change 'Set DATE-REF to '1858-11-17' from MJD-REF'.

In [14]: spec
Out[14]:
<Spectrum1D(flux=<Quantity [5., 6., 7.] Jy>, spectral_axis=<SpectralAxis
   (doppler_rest=1400000000.0 Hz
    doppler_convention=None)
  [6538.8, 6539.8, 6540.8] m>)>
```

Reply:
> This isn't strictly true: if a fits wcs is provided, the initializer will still attempt to create a `SpectralAxis` from the wcs information and fall back to a length unit if unsuccessful:

If the WCS finds a `CTYPE` it can interpret as length (or frequency). Otherwise, without `CTYPE1` or set to something like `'LINEAR'` in our example spectrum above, the default unit becomes `pixel`. But that is an accepted unit for `spectral_axis` as well (and results in a separate error when trying to call e.g. `spec.with_spectral_unit('nm')`), so I think this can be separated from the solution for #676.
Meaning I'd leave it to the loader to determine the best default unit for such a WCS.